api/envoy/api/v2/cluster/outlier_detection.proto

```diff
@@ -76,4 +76,14 @@ message OutlierDetection {
   // used to disable ejection or to ramp it up slowly. Defaults to 0.
   google.protobuf.UInt32Value enforcing_consecutive_gateway_failure = 11
       [(validate.rules).uint32.lte = 100];
+
+  // The number of consecutive connect failures before ejection
+  // occurs. Defaults to 5.
+  google.protobuf.UInt32Value consecutive_connect_failure = 12;
```

Comment:
Unfortunately I think this is more complicated than "connect failure" since some of the reasons for "connection failures" can be internal circuit breaking. We should make a more generic name here and clearly document how this is basically all errors that are internally generated by Envoy (failure to TCP connection, TLS handshake, overflow, timeout, etc.).

One special case that needs considering is when upstream actually *does* send an HTTP reset. How should this be handled? IMO it should count as an upstream real service unavailable error. Anyway needs thinking about.

Once we lock down the nomenclature here I think we will need to update the entire PR to use the new wording.

Reply:
> Unfortunately I think this is more complicated than "connect failure" since some of the reasons for "connection failures" can be internal circuit breaking. We should make a more generic name here and clearly document how this is basically all errors that are internally generated by Envoy (failure to TCP connection, TLS handshake, overflow, timeout, etc.).

@mattklein123 I can definitely address those cases. Outlier detector works on the principle that errors are reported to an outlier implementation attached to a host. I guess that other errors you mentioned: TCP connection, TLS handshake, overflow, timeout are currently not reported to host's outlier detector, correct? 

> We should make a more generic name...

What is your suggestion for naming those types of errors?

Reply:
> I guess that other errors you mentioned: TCP connection, TLS handshake, overflow, timeout are currently not reported to host's outlier detector, correct?

Yes they are reported. These generally come through as "resets" of various types.

> What is your suggestion for naming those types of errors?

I'm not sure of the best name, but something along the lines of "local origin error" vs. "remote origin error" or something like that is how I would think about it.

Reply:
Thanks. I will look at TCP connection and TLS handshake first and will come up with a proposal.

Reply:
If we are looking for names here, I would do something like: "local_origin" vs. "connect" and describe what that means. I think "local_origin" is a wrapper for circuit breaking, connect failure, etc.